src/subnet.c

```diff
@@ -264,17 +264,39 @@ subnet_t *lookup_subnet_ipv4(const ipv4_t *address) {
 
 	// Search all subnets for a matching one
 
+	int last_rtt = INT_MAX; // current smallest rtt seen
+
 	for splay_each(subnet_t, p, &subnet_tree) {
 		if(!p || p->type != SUBNET_IPV4) {
 			continue;
 		}
 
 		if(!maskcmp(address, &p->net.ipv4.address, p->net.ipv4.prefixlength)) {
-			r = p;
```

Comment:
I didn't notice this before, but by removing this line you've subtly changed how tinc handles packets routed to unreachable nodes. Before, `lookup_subnet_*()` would return a non-`NULL` result even if the subnet found wasn't reachable. Then tinc would send `ICMP_NET_UNREACH` packets back. But now you'll return `NULL`, so then tinc will send back `ICMP_NET_UNKNOWN`.